### PR TITLE
Feat: Add `OutOfStock` components

### PR DESCRIPTION
**Pull request by renatamottam**
## What's the purpose of this pull request?
`fs/components`
- [x] Moves `OutOfStock` component
- [x] Moves `OutOfStockTitle` component

`fs/ui`
- [x] Moves `OutOfStock` styles

`fs/core`
- [x] Review `OutOfStock` structure
- [x] Delete `OutOfStock`'s Storybook and styles files

#### You can test the usage of this component in the `faststore/core` application:
1. Run `yarn dev`
2. Remove the `!` from the `!availability &&` in line **197** of fs/core's `ProductDetails.tsx`
3. Check the `OutOfStock` feature on the store's PDP

Ps: `OutOfStock` integration is not implemented yet.

https://user-images.githubusercontent.com/3356699/228099488-1293a624-a728-45e3-a19d-02a4125ce440.mov

### Starters Deploy Preview
`WIP`

### CODE REVIEW

**Review by ArthurTriis1 — APPROVED** [review]

File: apps/site/pages/components/organisms/out-of-stock.mdx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+## Props
+
+### Out of Stock
+
+<OutOfStockPropsSection component="OutOfStock" />
+
+### Out of Stock Title
+
+export const propsOutOfStockTitle = [
```
Comment (eduardoformiga) [inline apps/site/pages/components/organisms/out-of-stock.mdx:103]: Have you ever tried changing the type from `OutOfStockTitleProps` for the interface? In my previous tests, types don't work as expected :(

File: packages/components/src/organisms/OutOfStock/OutOfStockTitle.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React, { ReactNode } from 'react'
+
+export type OutOfStockTitleProps = {
```
Comment (eduardoformiga) [inline packages/components/src/organisms/OutOfStock/OutOfStockTitle.tsx:3]: react-docgen-typescript works better with an interface to extract the props automatically.

```suggestion
export interface OutOfStockTitleProps = {
```

File: packages/core/src/components/product/OutOfStock/OutOfStock.tsx [hunk 3]
```diff
@@ -1,133 +1,52 @@
...
-  }
+  const { pushToast } = useUI()
 
   const handleSubmit = (event: FormEvent) => {
     event.preventDefault()
+    setIsLoading(true)
 
-    setDisabled(true)
-    setButtonIconName('Ellipsis')
+    // TODO: Missing integration
```
Comment (eduardoformiga) [inline packages/core/src/components/product/OutOfStock/OutOfStock.tsx:19]: 👀

File: packages/core/src/components/product/OutOfStock/OutOfStock.tsx [hunk 4]
```diff
@@ -1,133 +1,52 @@
...
+
+      const formElement = event.currentTarget as HTMLFormElement
+      formElement.reset()
+
+      pushToast({
+        title: 'Subscribed successfully!',
+        message: "You'll be notified when this product is back to stock.",
+        status: 'INFO',
+        icon: <Icon name="CircleWavyCheck" width={30} height={30} />,
+      })
```
Comment (eduardoformiga) [inline packages/core/src/components/product/OutOfStock/OutOfStock.tsx:37]: This block of code in `finally` block will be executed even after the `catch` (error). Should we place it in the `try {...}` block?

File: packages/components/src/organisms/OutOfStock/OutOfStockTitle.tsx
Comment (eduardoformiga) [inline packages/components/src/organisms/OutOfStock/OutOfStockTitle.tsx:1]: I wonder if this component makes sense or if we should incorporate it inside the `OutOfStock` code. We are not using it directly from the core, right?

**Review by eduardoformiga — COMMENTED** [review]
Woww, Good job, Fan. 

Lots of hidden tasks and refactors were done in this PR 🎉 .
I left some comments along with the review.

File: packages/core/src/components/product/OutOfStock/OutOfStock.tsx (hunk 4 above)
Comment (hellofanny) [reply packages/core/src/components/product/OutOfStock/OutOfStock.tsx:37]: oh that's right! Moving it!

File: apps/site/pages/components/organisms/out-of-stock.mdx (hunk 1 above)
Comment (hellofanny) [reply apps/site/pages/components/organisms/out-of-stock.mdx:103]: Lemme try! :)

File: packages/components/src/organisms/OutOfStock/OutOfStockTitle.tsx
Comment (hellofanny) [reply packages/components/src/organisms/OutOfStock/OutOfStockTitle.tsx:1]: Yes! I think it should be incorporated to `OutOfStock`, I don't see any other reason to keep it compound.

Comment (hellofanny) [reply packages/components/src/organisms/OutOfStock/OutOfStockTitle.tsx:1]: Actually, I think this component should be the `Heading` component.
I just found that we had an [old](https://vtex-dev.atlassian.net/browse/FS-405?atlOrigin=eyJpIjoiYjRhMzVjYmIyYzRlNDE1Nzk5YjE5ZmQ2NTQ1YWY1YzIiLCJwIjoiaiJ9) task for it.

Comment (hellofanny) [reply packages/components/src/organisms/OutOfStock/OutOfStockTitle.tsx:1]: I decided to remove `OutOfStockTitle` for now.

File: apps/site/pages/components/organisms/out-of-stock.mdx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+  </Tab>
+</Tabs>
+
+---
+
+## Props
+
+### Out of Stock
+
+<OutOfStockPropsSection component="OutOfStock" />
```
Comment (eduardoformiga) [inline apps/site/pages/components/organisms/out-of-stock.mdx:99]: 🎉

**Review by eduardoformiga — APPROVED** [review]

**Pull request merged**